src/main/java/repository/FileRepository.java

```diff
@@ -0,0 +1,70 @@
+package repository;
+
+import model.Book;
+import util.CsvFileUtil;
+
+import java.util.*;
+
+public class FileRepository implements Repository {
+
+    private final CsvFileUtil csvFileUtil;
+    Map<Long, Book> bookMap = new HashMap<>();
+
+    public FileRepository(String file_path) {
+        csvFileUtil = new CsvFileUtil(file_path);
+        bookMap = csvFileUtil.readAllBooksFromCsv();
+    }
+
+    // 책 등록
+    @Override
+    public void saveBook(Book book) {
+        bookMap.put(book.getBookNo(), book);
+        csvFileUtil.writeFile(bookMap);
+    }
+
+    // 모든 book 반환
+    @Override
+    public List<Book> findAllBook() {
+        return bookMap.values().stream()
+                .toList();
+    }
+
+    // title에 특정 문자가 포함된 Book 객체를 모두 찾는 메서드
+    @Override
+    public List<Book> findBookByTitle(String searchTitle) {
+        return bookMap.values().stream()
+                .filter(book -> book.getTitle().contains(searchTitle))
```

Comment:
```suggestion
book.getTitle().contains(searchTitle)
```
이렇게 한줄의 점 여러개 있는 경우를 `디미터의 법칙` 이라고 합니다.
대표적으로 객체지향을 지킬수 있는 아주 좋은 법칙인데요.

이런경우 book에게 메시지를 보내서 알려달라고 하면 어떨까요?